tests/unit/plugins/inventory/test_digitalocean.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2021 Ansible Project
+# GNGeneral Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import pytest
+
+from ansible.errors import AnsibleParserError
+from ansible.inventory.data import InventoryData
+from ansible_collections.community.digitalocean.plugins.inventory.digitalocean import InventoryModule
+
+
+@pytest.fixture(scope="module")
+def inventory():
+    r = InventoryModule()
+    r.inventory = InventoryData()
+    return r
+
+
+def test_verify_file_bad_config(inventory):
+    assert inventory.verify_file('digitalocean_foobar.yml') is False
+
+
+def test_validate_config(inventory):
```

Comment:
Each test function should only have one set of assert statements. I would suggest that you split the `test_validate_config` into three separate functions named `test_validate_config_invalid_attribute`, `test_validate_config_invalid_prefix`, and `test_validate_config_invalid_pagination`. You may also want to consider adding a test on a valid data.

Reply:
I'd personally remove the validation code from the plugin, and thus also the tests for them :)

Reply:
Regarding pagination and attributes I am happily do so, but is it safe to leave the user to set whatever var_prefix (e.g. 'my do-')?

Reply:
`my do-` can also be used, for example `hostvars[inventory_hostname]['my do-status']`. That's not recommended, but possible.

Reply:
I believe you and not bond to it emotionally. If it's ok for @tadeboro as well, I wipe out the whole section.